FhirToDataLake/src/Microsoft.Health.Fhir.Synapse.JobManagement/AzureStorageQueueClient.cs

```diff
@@ -0,0 +1,675 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System.Diagnostics;
+using Azure;
+using Azure.Data.Tables;
+using Azure.Identity;
+using Azure.Storage.Queues;
+using Azure.Storage.Queues.Models;
+using EnsureThat;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Health.Fhir.Synapse.Common.Configurations;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Exceptions;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Extensions;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Models;
+using Microsoft.Health.Fhir.Synapse.JobManagement.Models.AzureStorage;
+using Microsoft.Health.JobManagement;
+using Newtonsoft.Json;
+
+namespace Microsoft.Health.Fhir.Synapse.JobManagement
+{
+    public class AzureStorageQueueClient<TJobInfo> : IQueueClient
+        where TJobInfo : AzureStorageJobInfo, new()
+    {
+        private readonly TableClient _azureJobInfoTableClient;
+
+        private readonly QueueClient _azureJobMessageQueueClient;
+
+        private readonly ILogger<AzureStorageQueueClient<TJobInfo>> _logger;
+
+        private readonly int _maximumSizeOfAnEntityInBytes = 1024 * 1024;
+
+        public AzureStorageQueueClient(
+            IStorage storage,
+            IOptions<JobConfiguration> jobConfiguration,
+            ILogger<AzureStorageQueueClient<TJobInfo>> logger)
+        {
+            EnsureArg.IsNotNull(jobConfiguration, nameof(jobConfiguration));
+            EnsureArg.IsNotNull(storage, nameof(storage));
+
+            _logger = EnsureArg.IsNotNull(logger, nameof(logger));
+
+            if (storage.UseConnectionString)
+            {
+                _azureJobInfoTableClient = new TableClient(storage.TableUrl, storage.TableName);
+                _azureJobMessageQueueClient = new QueueClient(storage.QueueUrl, storage.QueueName);
+            }
+            else
+            {
+                _azureJobInfoTableClient = new TableClient(
+                    new Uri(storage.TableUrl),
+                    storage.TableName,
+                    new DefaultAzureCredential());
+                _azureJobMessageQueueClient = new QueueClient(
+                    new Uri($"{storage.QueueUrl}{storage.QueueName}"),
+                    new DefaultAzureCredential());
+            }
+        }
+
+        public bool IsInitialized()
```

Comment:
Maybe renamed to ```TryInitialize```?

Reply:
`IsInitialized` is a interface designed by `IQueueClient` in [TaskManagement](https://github.com/microsoft/fhir-server/blob/4491396b046c20b834a92e92141a5d607e408804/src/Microsoft.Health.TaskManagement/IQueueClient.cs#L17)  library, and we can't rename it.